### DISCUSSION TITLE
Path resolver for MayaUsd

### DISCUSSION THREAD

**Discussion by silviapalara** [Autodesk/maya-usd#3147] [Q&A]
Hello, I have a path resolver we have been using with USD standalone and Houdini and I'd love to adapt it for Maya too. 
For Houdini I built it against their installation of USD and hython and I was wondering if I should do something similar for Maya? MayaUSD is built against the standalone USD libraries, can I use those and mayapy? Do I need the Maya devkit? 
Also once the plugin is built, will Maya pick it up through the PXR_PLUGINPATH_NAME env var or do I need to turn it into a Maya plugin?

I have Maya 2023 and would like to build it for Windows 10 and CentOS.

Thank you for any help you can give me.

**Comment by gelshoffNV** [comment]
You can build against the USD delivered with maya-usd, and set the PXR_PLUGINPATH_NAME before Maya starts, either a plugin .mod file or userSetup.py would work. You don't need any maya devkit unless you want to resolve outside of USD, like a maya file resolver. You can use usd python libs without the resolver, just with maya-usd, you can do `from pxr import Usd`, etc. If you set PXR_PLUGINPATH_NAME and have your plugInfo.json and dll /so correct, maya, or usd really, will use your resolver. If you're using the maya-usd proxy node, though, you'll need to set the path string directly to your uri, since the only ui's connected are FileBrowsers which won't be connected to the custom uris.
